pkg/wrangler/context.go

```diff
@@ -262,10 +290,31 @@ func (w *Context) WithAgent(userAgent string) *Context {
 	wContextCopy.API = wContextCopy.api.WithAgent(userAgent).V1()
 	wContextCopy.CRD = wContextCopy.crd.WithAgent(userAgent).V1()
 	wContextCopy.Plan = wContextCopy.plan.WithAgent(userAgent).V1()
+	if w.DeferredCAPIRegistration.CAPIInitialized {
+		wContextCopy.CAPI = wContextCopy.capi.WithAgent(userAgent).V1beta1()
+	} else {
+		go wContextCopy.BackfillCAPI(&wContextCopy)
+	}
 
 	return &wContextCopy
 }
 
+func (w *Context) BackfillCAPI(cpy *Context) {
```

Comment:
This is more of a nit. The function seems to be coded to backfill `w` from the `cpy`.
Because of these semantics I would name the argument the `origin`, here and where used in the function.
```suggestion
func (w *Context) BackfillCAPI(origin *Context) {
```